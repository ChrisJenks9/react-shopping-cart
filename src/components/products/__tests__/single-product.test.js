import React from 'react';
import { shallow, mount } from 'enzyme';
import Database from '../../../db.json';
import SingleProduct from '../single-product';

describe('#SingleProduct()', () => {
  test('renders with a single product without crashing', () => {
    const singleProduct = shallow(<SingleProduct key={Database.products[0].id} item={Database.products[0]} />);
    expect(singleProduct).toMatchSnapshot();
  });

  describe('inStock()', () => {
    test('To return add to cart button if there is > 0 stock', () => {
      const singleProduct = mount(<SingleProduct key={Database.products[0].id} item={Database.products[0]} />);
      expect(singleProduct.instance().inStock()).toMatchSnapshot();
    });
    test('To return out of stock button if there is 0 stock', () => {
      const singleProduct = mount(<SingleProduct key={Database.products[4].id} item={Database.products[4]} />);
      expect(singleProduct.instance().inStock()).toMatchSnapshot();
    });
  });
});
