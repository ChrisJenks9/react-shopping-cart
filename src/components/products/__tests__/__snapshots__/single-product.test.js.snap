// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SingleProduct() inStock() To return add to cart button if there is > 0 stock 1`] = `
<Button
  className=""
  color="info"
  onClick={[Function]}
  tag="button"
>
  ADD TO CART
</Button>
`;

exports[`#SingleProduct() inStock() To return out of stock button if there is 0 stock 1`] = `
<Button
  color="warning"
  tag="button"
>
  Out Of Stock
</Button>
`;

exports[`#SingleProduct() renders with a single product without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleProduct
    item={
        Object {
            "category-id": 1,
            "description": "At vero eos et accusamus et iusto odio dignissimos.",
            "id": 1,
            "image": "http://hranew.com/wp-content/uploads/2015/04/shop-placeholder.png",
            "price": 99,
            "stock": 5,
            "title": "Almond Toe Court Shoes, Patent Black",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h6
          className="product-title"
>
          Almond Toe Court Shoes, Patent Black
</h6>,
        <div
          className="product-image"
>
          <img
                    alt=""
                    src="http://hranew.com/wp-content/uploads/2015/04/shop-placeholder.png"
          />
</div>,
        <p
          className="product-description"
>
          At vero eos et accusamus et iusto odio dignissimos.
</p>,
        <h6
          className="product-price"
>
          £
          99
</h6>,
        <div
          className="product-add-to-cart"
>
          <Unknown />
</div>,
      ],
      "className": "single-product",
      "lg": "3",
      "md": "4",
      "tag": "div",
      "widths": Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ],
      "xs": "12",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Almond Toe Court Shoes, Patent Black",
          "className": "product-title",
        },
        "ref": null,
        "rendered": "Almond Toe Court Shoes, Patent Black",
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            src="http://hranew.com/wp-content/uploads/2015/04/shop-placeholder.png"
/>,
          "className": "product-image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "http://hranew.com/wp-content/uploads/2015/04/shop-placeholder.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "At vero eos et accusamus et iusto odio dignissimos.",
          "className": "product-description",
        },
        "ref": null,
        "rendered": "At vero eos et accusamus et iusto odio dignissimos.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "£",
            99,
          ],
          "className": "product-price",
        },
        "ref": null,
        "rendered": Array [
          "£",
          99,
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown />,
          "className": "product-add-to-cart",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h6
            className="product-title"
>
            Almond Toe Court Shoes, Patent Black
</h6>,
          <div
            className="product-image"
>
            <img
                        alt=""
                        src="http://hranew.com/wp-content/uploads/2015/04/shop-placeholder.png"
            />
</div>,
          <p
            className="product-description"
>
            At vero eos et accusamus et iusto odio dignissimos.
</p>,
          <h6
            className="product-price"
>
            £
            99
</h6>,
          <div
            className="product-add-to-cart"
>
            <Unknown />
</div>,
        ],
        "className": "single-product",
        "lg": "3",
        "md": "4",
        "tag": "div",
        "widths": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "xs": "12",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Almond Toe Court Shoes, Patent Black",
            "className": "product-title",
          },
          "ref": null,
          "rendered": "Almond Toe Court Shoes, Patent Black",
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              src="http://hranew.com/wp-content/uploads/2015/04/shop-placeholder.png"
/>,
            "className": "product-image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "http://hranew.com/wp-content/uploads/2015/04/shop-placeholder.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "At vero eos et accusamus et iusto odio dignissimos.",
            "className": "product-description",
          },
          "ref": null,
          "rendered": "At vero eos et accusamus et iusto odio dignissimos.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "£",
              99,
            ],
            "className": "product-price",
          },
          "ref": null,
          "rendered": Array [
            "£",
            99,
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown />,
            "className": "product-add-to-cart",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
